internal/bucket/versioning/versioning.go

```diff
@@ -33,24 +34,48 @@ const (
 	Suspended State = "Suspended"
 )
 
+var (
+	errExcludedPrefixNotSupported = Errorf("excluded prefixes extension supported only when versioning is enabled")
+	errTooManyExcludedPrefixes    = Errorf("too many excluded prefixes")
+)
+
 // Versioning - Configuration for bucket versioning.
 type Versioning struct {
 	XMLNS   string   `xml:"xmlns,attr,omitempty"`
 	XMLName xml.Name `xml:"VersioningConfiguration"`
 	// MFADelete State    `xml:"MFADelete,omitempty"` // not supported yet.
 	Status State `xml:"Status,omitempty"`
+	// MinIO extension - allows selective, prefix-level versioning exclusion.
+	// Requires versioning to be enabled
+	ExcludedPrefixes []string `xml:">Prefix,omitempty"`
 }
 
 // Validate - validates the versioning configuration
 func (v Versioning) Validate() error {
+
 	// Not supported yet
 	// switch v.MFADelete {
 	// case Enabled, Disabled:
 	// default:
 	// 	return Errorf("unsupported MFADelete state %s", v.MFADelete)
 	// }
 	switch v.Status {
-	case Enabled, Suspended:
+	case Enabled:
+		const maxExcludedPrefixes = 10
+		if len(v.ExcludedPrefixes) > maxExcludedPrefixes {
+			return errTooManyExcludedPrefixes
+		}
+		for _, sprefix := range v.ExcludedPrefixes {
+			// Use filepath.Match to check for bad patterns
+			if _, err := filepath.Match(sprefix, ""); err != nil {
```

Comment:
Don't use filepath.Match() always used wildcard.Match() for similar AWS S3 semantics